testing-pre-releases.md

```diff
@@ -0,0 +1,264 @@
+# Testing pre-release versions of Kubernetes with kubeadm
+
+## Overview
+
+When testing Kubernetes with kubeadm three different version numbers should be considered:
+
+- The version of .deb or .rpm packages, including kubelet, kubectl and related dependencies (e.g. kubernetes-cni);
+  it is convenient to include also kubeadm package in this group to get the kubelet drop-in file deployed automatically.
+- The version of the kubeadm binary to be used, that can eventually overwrite the kubeadm binary included in
+  the .deb or .rpm packages. e.g. Use v1.10.3 GA packages for kubelet, kubectl, but test a local build of kubernetes.
+- The `--kubernetes-version` flag of kubeadm or the `kubernetesVersion` field of Kubeadm API,
+  defining the version number of the control plane components installed by kubeadm.
+
+For each of the above set version number, it is also necessary to understand the supported version skew policy,
+the availability of pre-compiled artifacts, or, as alternative, how to create a local version through the build process.
+
+<!-- TOC -->
+
+- [Testing pre-release versions of Kubernetes with kubeadm](#testing-pre-release-versions-of-kubernetes-with-kubeadm)
+    - [Overview](#overview)
+    - [Kubeadm version skew policy](#kubeadm-version-skew-policy)
+    - [Availability of pre-compiled release artifacts](#availability-of-pre-compiled-release-artifacts)
+        - [Getting .deb or .rpm packages form repository](#getting-deb-or-rpm-packages-form-repository)
+        - [Getting .deb or .rpm packages form a GCS bucket](#getting-deb-or-rpm-packages-form-a-gcs-bucket)
+        - [Getting kubeadm binaries from a GCS bucket](#getting-kubeadm-binaries-from-a-gcs-bucket)
+        - [Getting docker images from a GCR registry](#getting-docker-images-from-a-gcr-registry)
+    - [Create a local version](#create-a-local-version)
+        - [Build .debs packages](#build-debs-packages)
+        - [Build kubeadm binary](#build-kubeadm-binary)
+        - [Build controlplane docker images](#build-controlplane-docker-images)
+    - [Tips and Tricks](#tips-and-tricks)
+        - [Semantic version ordering](#semantic-version-ordering)
+        - [How to identify exact version/build number for a PR](#how-to-identify-exact-versionbuild-number-for-a-pr)
+        - [Change the target version number when building a local release](#change-the-target-version-number-when-building-a-local-release)
+
+<!-- /TOC -->
+
+## Kubeadm version skew policy
+
+The [Kubeadm version skew policy](https://kubernetes.io/docs/setup/independent/create-cluster-kubeadm/#version-skew-policy)
+can briefly be summarized in:
+
+- The kubeadm CLI tool of version vX.Y may deploy clusters with a control plane of version vX.Y or vX.(Y-1).
+- The kubeadm CLI vX.Y can upgrade an existing kubeadm-created cluster of version vX.(Y-1).
+- One minor version skew between the kubelet and the control plane is supported, but the kubelet version may never
```

Comment:
might want to spell out major.minor.patch, I always get the latter two confused.